pyiceberg/io/pyarrow.py

```diff
@@ -1238,10 +1238,12 @@ def _task_to_record_batches(
         for batch in batches:
             next_index = next_index + len(batch)
             current_index = next_index - len(batch)
+            output_batches = iter([batch])
             if positional_deletes:
                 # Create the mask of indices that we're interested in
                 indices = _combine_positional_deletes(positional_deletes, current_index, current_index + len(batch))
                 batch = batch.take(indices)
+
                 # Apply the user filter
                 if pyarrow_filter is not None:
```

Comment:
a little unrelated to this PR, but I wonder if there's ever a case where we need to filter (with `pyarrow_filter`) but `positional_deletes` is None/empty/falsey. 
I can see that as a potential issue

Reply:
Great question! That's already handled here: 

https://github.com/apache/iceberg-python/blob/0dc54080aa287dc8e920da128d7f4b335965f1df/pyiceberg/io/pyarrow.py#L1221-L1234

Reply:
thanks! 
So if there are positional_deletes, apply filter manually to the pyarrow table. 
If there are no positional_deletes, pushdown the filter to scanner. 

arrow 17 should make this a lot cleaner